App/Classes/Services/UserServices/UserInteractionService.cs

```diff
@@ -366,22 +366,31 @@ public async Task<List<User>> GetAllBlockedBy()
     }
 
     public List<IInteractionSubscriber> Subscribers { get; } = [];
+    private SemaphoreSlim SubscribersLock { get; } = new(1, 1);
```

Comment:
the semaphore here doesn't appear to work in preventing errors when the subscriber list is modified - can see that an error is thrown when unfriending someone while in the dm view for them